src/wp-includes/html-api/class-wp-html-tag-processor.php

```diff
@@ -711,6 +711,41 @@ public function release_bookmark( $name ) {
 	}
 
 
+	/**
+	 * Allow internal WordPress code to extract raw HTML snippets associated with a matched tag.
+	 *
+	 * This is meant for internal use to support existing filters that pass segments of input HTML
+	 * for further analysis. Filters which rely on these values should be updated to interact with
+	 * the structural methods this class provides.
+	 *
+	 * @access private
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param string $internal_use_only Required to accept liability for exposing system internals through this function.
+	 * @param string $content           What portion of the raw HTML markup to return: 'entire-tag' or 'only-attributes'.
+	 * @return string|null Raw HTML markup from input document for matched tag, if matched, else NULL.
+	 */
+	public function _wp_internal_extract_raw_token_markup( $internal_use_only, $contents = 'entire-tag' ) {
+		$disclaimer = <<<INTERNAL_ONLY
+			I understand that this is only for internal WordPress usage and something
+			will likely break if used elsewhere. This function comes with no warranty.
+INTERNAL_ONLY;
+
+		if ( $internal_use_only !== $disclaimer || null === $this->tag_name_starts_at ) {
```

Comment:
Is it only necessary to check for `tag_name_starts_at`? If that property is set, does it follow that `tag_ends_at` is set too? If not, we could add the check here, but if it does, adding a clarifying (assertion) comment would improve code readability.

Reply:
that's correct, and as internal values this isn't exposed. it's an idiom used throughout the Tag Processor. I've thought about different means to check what we're doing, but I don't think adding the same comment to twenty different places within the class would be that useful.

I do agree this is a bit indirect; more or less though I'm waiting until a much clearer and better solution appears.